### PR TITLE
ecl: Update To 21.2.1 

### PR DESCRIPTION
**Pull request by CamberLoid**
Topic Description
-----------------

The topic will upgrade the embeddable common lisp compiler `ecl` to version 21.2.1, along with the update of maxima and wxmaxima. 
The update of `ecl` brings major so version change.

Note: ecl does not come with a integrated test. So as long as maxima is successfully build, ecl will be considered functional,

Package(s) Affected
-------------------

Update: `ecl` `maxima` `wxmaxima`
Rebuild: `texlive`

Security Update?
----------------

No

Build Order
-----------

`ciel build -i $INSTANCE ecl maxima wxmaxima texlive`

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: extra-lisp/ecl/autobuild/defines [hunk 1]
```diff
@@ -8,8 +8,7 @@ NOPARALLEL=1
 AUTOTOOLS_AFTER="--enable-unicode=yes \
                  --enable-c99complex \
                  --enable-threads=yes \
-                 --with-__thread \
-                 --disable-rpath \
+                 --enable-rpath=no \
```
Comment (MingcongBai) [inline extra-lisp/ecl/autobuild/defines]: That shouldn't have made a difference.

File: extra-lisp/ecl/autobuild/prepare [hunk 2]
```diff
@@ -4,7 +4,6 @@ find src/c -type f -perm /0111 | xargs chmod a-x
 find src/h -type f -perm /0111 | xargs chmod a-x
 
 sed -i 's/{.*,.*,.*,.*,.*}/{&}/g' src/c/symbols_list.h
```
Comment (MingcongBai) [inline extra-lisp/ecl/autobuild/prepare:10]: Could you please add abinfo to these files?

File: extra-scientific/maxima/autobuild/beyond [hunk 3]
```diff
@@ -1,2 +1,2 @@
 # FIXME: fixed version
-install -Dm755 src/binary-ecl/maxima.fas "$PKGDIR"/usr/lib/ecl-16.1.3/maxima.fas
```
Comment (MingcongBai) [inline extra-scientific/maxima/autobuild/beyond:2]: Best to use the PKGVER variable.

File: extra-scientific/maxima/autobuild/defines [hunk 4]
```diff
@@ -4,9 +4,22 @@ PKGDEP="gnuplot texinfo ecl tk rlwrap"
...
 
-AUTOTOOLS_AFTER="--enable-ecl --with-default-lisp=ecl \
-                 --libexecdir=/usr/lib"
+AUTOTOOLS_AFTER="""
+    --prefix=/usr
+    --mandir=/usr/share/man
+    --infodir=/usr/share/info
+    --enable-ecl 
+    --with-default-lisp=ecl 
+    --libexecdir=/usr/lib
```
Comment (MingcongBai) [inline extra-scientific/maxima/autobuild/defines]: We don't use this convention, remove this line.

File: extra-scientific/maxima/autobuild/defines [hunk 5]
```diff
@@ -4,9 +4,22 @@ PKGDEP="gnuplot texinfo ecl tk rlwrap"
...
+    """
+
 ABSHADOW=no
-MAKE_AFTER="emacsdir=/usr/share/emacs/site-lisp/maxima"
+RECONF=0
 NOPARALLEL=1
 NOLTO=1
+
+ABSPLITDBG=0
```
Comment (MingcongBai) [inline extra-scientific/maxima/autobuild/defines:18]: Why?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-scientific/maxima/autobuild/beyond (hunk 3 above)
Comment (Icenowy) [reply extra-scientific/maxima/autobuild/beyond:2]: Well this is ECL's version, not Maxima's.

File: extra-scientific/maxima/autobuild/defines (hunk 5 above)
Comment (Icenowy) [reply extra-scientific/maxima/autobuild/defines:18]: > Why?

It's built by ECL, so no meaningful debugging info.

File: extra-scientific/maxima/autobuild/beyond (hunk 3 above)
Comment (CamberLoid) [reply extra-scientific/maxima/autobuild/beyond:2]: The version here is not maxima's version, therefore PKGVER variable is not applicable in this scenario,

File: extra-lisp/ecl/autobuild/defines (hunk 1 above)
Comment (CamberLoid) [reply extra-lisp/ecl/autobuild/defines]: - with-__theard: deprecated.
- enable-rpath=no: The change is meant to keep with upstream's expression.

So yes, not make a difference.

**Comment by Icenowy** [comment]
@CamberLoid Could you squash changes to one commit per package? (Personal recommendation)

**Review by Icenowy — APPROVED** [review]

**Review by MingcongBai — APPROVED** [review]

**Comment by CamberLoid** [comment]
https://gist.github.com/CamberLoid/881e9f29a9b7885c6dff516d0523e761
https://gist.github.com/CamberLoid/82f6d3581f30828a8df40b2f7dc5aef2

**Pull request merged**